### PR TITLE
Use new grant interface in mirage-xen

### PR DESCRIPTION
**Pull request by yomimono**

### CODE REVIEW

File: xen/console_xen.ml [hunk 1]
```diff
@@ -134,7 +133,7 @@ let get_initial_console () =
   (* The domain is created with a reserved grant entry already set up.
      We don't need to know who the backend domain is. *)
   let backend_id = 0 in (* unused *)
-  let gnt = Gnt.console in (* unused *)
+  let gnt = Xen.console in (* unused *)
```
Comment (talex5) [inline xen/console_xen.ml]: Any idea why these lines have `(* unused *)` at the end of them?

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ script: bash -ex .travis-opam.sh
 dist: trusty
 env:
   global:
-  - PINS="mirage-console:. mirage-console-xen:. mirage-console-unix:. mirage-console-xen-backend:."
+  - "mirage-xen mirage-console:. mirage-console-xen:. mirage-console-unix:. mirage-console-xen-backend:."
```
Comment (talex5) [inline .travis.yml]: I don't think this needs pinning. It just needs a version constraint in the opam.

File: mirage-console-xen-backend.opam [hunk 3]
```diff
@@ -11,11 +11,11 @@ depends: [
   "dune" {build & >= "1.0"}
   "mirage-console-lwt" {>= "2.2.0"}
   "mirage-console-xen-proto"
+  "mirage-xen"
```
Comment (talex5) [inline mirage-console-xen-backend.opam]: `>= 3.3.0` ?

**Comment by avsm** [comment]
If I remember, you want to be a little careful with freeing the default console as dom0's xenconsoled used to get rather unhappy if it was closed before the domain disappeared. I haven't tested this on modern xenconsoled though.

File: .travis.yml (hunk 2 above)
Comment (yomimono) [reply .travis.yml]: Right you are; I'd forgotten that I released this...

File: xen/console_xen.ml (hunk 1 above)
Comment (yomimono) [reply xen/console_xen.ml]: Previously, no other code cared about them (write and read code refers only to `t.evtchn` and `t.ring`) but this is indeed no longer the case for `gnt` and `id`.

**Pull request merged**